### PR TITLE
Added score data display for milestone reports

### LINKED ISSUE #2245: Add score data display for milestone reports
Add a `<milestone-report-data>` AngularJS component that allows authors to display different calculations for a particular milestone subscore in the report content. Supported calculations will include:
- **percent**: a numerical percentage (e.g. '80%') of the class that received any of the specified score values for a particular subscore id
- **count**: the total number (e.g. '10') of students in the class that received any of the specified score values for a particular subscore id
- **total**: the total number (e.g. '20') of scored responses for a specified subscore id
- **average**: the class average score (e.g '3.2') 
for a specified subscore id

### PR DESCRIPTION
**Pull request by breity**
Added a new AngularJS component (MilestoneReportData) that shows basic calculations for a specific subscore id.

The component uses up to 3 attributes to specify binding values:
1. `score-id`, a string that corresponds to the c-rater subscore id for which you want to show data (required)
1. `calc`, a string that represents the type of calculation you want to perform on the data (required). Options are:
   - 'percent': shows the percentage of students that have scores matching one of the `score-values` values
   - 'count': shows the total number of students that have scores matching one of the `score-values` values
   - 'average': shows the class' average score for the specific subscore id
   - 'total': shows the total number of responses for the milestone item
1. `score-values`: Number array representing the scores that you want to match for the subscore id (required for 'percent' and 'count' `calc` values).

To test, set up a run with a milestone report. Add each of the following (or similar variations depending on the subscore ids in the c-rater item you are testing) to a milestone report's `content` field and make sure the requested data is display correctly when the teacher views the report:
- `<p><milestone-report-data score-id=\"ki\" calc=\"percent\" score-values=\"[4,5]\"></milestone-report-data> of students scored 4 or 5 for KI.</p>`
- `<p><milestone-report-data score-id=\"science\" calc=\"percent\" score-values=\"[3]\"></milestone-report-data> of students scored 3 for the science subscore.</p>`
-`<p><milestone-report-data score-id=\"experimentation\" calc=\"count\" score-values=\"[1,2]\"></milestone-report-data> student(s) scored 2 or lower for the experimentation subscore.</p>`
- `<p>The class average KI score is <b><milestone-report-data score-id=\"ki\" calc=\"average\"></milestone-report-data></b>.</p>`
- `<p><milestone-report-data score-id=\"ki\" calc=\"total\"></milestone-report-data> student(s) have submitted an answer for this item.</p>`

Resolves #2245.

### CODE REVIEW

File: src/main/webapp/wise5/classroomMonitor/milestones/milestonesController.js [hunk 1]
```diff
@@ -493,6 +463,51 @@ class MilestonesController {
     };
   }
 
+  processMilestoneGraphsAndData(content, aggregateAutoScores) {
+    for (let componentId of Object.keys(aggregateAutoScores)) {
+      const componentAggregate = aggregateAutoScores[componentId];
```
Comment (geoffreykwan) [inline src/main/webapp/wise5/classroomMonitor/milestones/milestonesController.js]: Can this be changed to
```
    for (const componentAggregate of aggregateAutoScores) {
```

File: src/main/webapp/wise5/classroomMonitor/milestones/milestonesController.js [hunk 2]
```diff
@@ -493,6 +463,51 @@ class MilestonesController {
...
+      let subScoreIndex = 0;
+      for (let subScoreId of Object.keys(componentAggregate)) {
```
Comment (geoffreykwan) [inline src/main/webapp/wise5/classroomMonitor/milestones/milestonesController.js]: Can this be changed to
```
for (let [subScoreId, aggregateData] of Object.entries(componentAggregate)) {
```

File: src/main/webapp/wise5/directives/milestoneReportData/milestoneReportData.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        };
+        if (this.calc === 'count') {
+            this.output = this.getCount();
+        };
+        if (this.calc === 'total') {
+            this.output = this.data.scoreCount;
+        };
+        if (this.calc === 'average') {
+            this.output = this.data.average;
+        };
```
Comment (geoffreykwan) [inline src/main/webapp/wise5/directives/milestoneReportData/milestoneReportData.js]: The semi colons after the if blocks can be removed.

**Comment by breity** [comment]
Thanks @geoffreykwan. I made the requested changes and removed the 'total' calculation option. 'count' with no 'scoreValues' now shows the total responses.

**Pull request merged**